File: PYTHON/Program/Motor.py
from Kendaraan import Kendaraan

class Motor(Kendaraan):
    __JenisMotor = ""
    __NamaMotor = ""
    def __init__(self, JenisMotor, NamaMotor, platnomor = "", merk = "", tahun = 2024, warna = "", cckendaraan = 110, tangki = 5):
        super().__init__(platnomor, merk, tahun, warna, cckendaraan, tangki)
        self.JenisMotor = JenisMotor
        self.NamaMotor = NamaMotor
    
    def get_JenisMotor(self):
        return self.JenisMotor

    def set_JenisMotor(self, JenisMotor):
        self.JenisMotor = JenisMotor
        
    def get_NamaMotor(self):
        return self.NamaMotor

    def set_NamaMotor(self, NamaMotor):
        self.NamaMotor = NamaMotor
    
    def identifikasi(self):
        identifikasi_kendaraan = super().identifikasi()
        return f"{self.JenisMotor}\n- Nama Motor :{self.NamaMotor}\n- {identifikasi_kendaraan}"

File: CPP/Program/Motor.cpp
#include <iostream>
#include <string>
#include "Kendaraan.cpp"

#pragma once

class Motor : public Kendaraan {
private:
    std::string JenisMotor;
    std::string NamaMotor;

public:
    Motor(std::string JenisMotor, std::string NamaMotor,
          std::string platnomor = "", std::string merk = "", int tahun = 2024, std::string warna = "", int cckendaraan = 110, int tangki = 5)
        : Kendaraan(platnomor, merk, tahun, warna, cckendaraan, tangki), JenisMotor(JenisMotor), NamaMotor(NamaMotor) {}

    std::string get_JenisMotor() const {
        return JenisMotor;
    }

    void set_JenisMotor(const std::string& JenisMotor) {
        this->JenisMotor = JenisMotor;
    }

    std::string get_NamaMotor() const {
        return NamaMotor;
    }

    void set_NamaMotor(const std::string& NamaMotor) {
        this->NamaMotor = NamaMotor;
    }

    void identifikasi() const override{
    std::cout << JenisMotor << "\n- Nama Motor : " << NamaMotor << "\n";
    Kendaraan::identifikasi();  
}
};
